iroh-net/src/magicsock/udp_conn.rs

```diff
@@ -73,21 +119,63 @@ impl AsyncUdpSocket for UdpConn {
         meta: &mut [quinn_udp::RecvMeta],
     ) -> Poll<io::Result<usize>> {
         loop {
-            ready!(self.io.poll_recv_ready(cx))?;
-            if let Ok(res) = self.io.try_io(Interest::READABLE, || {
-                self.inner.recv(Arc::as_ref(&self.io).into(), bufs, meta)
-            }) {
-                for meta in meta.iter().take(res) {
-                    trace!(
-                        src = %meta.addr,
-                        len = meta.len,
-                        count = meta.len / meta.stride,
-                        dst = %meta.dst_ip.map(|x| x.to_string()).unwrap_or_default(),
-                        "UDP recv"
-                    );
+            if self.io.is_broken() {
+                match self.rebind() {
+                    Ok(()) => {
+                        // all good
+                    }
+                    Err(err) => {
+                        warn!("failed to rebind socket: {:?}", err);
+                        // TODO: improve error
+                        let err =
+                            std::io::Error::new(std::io::ErrorKind::NotConnected, err.to_string());
```

Comment:
I fear that quinn will swallow these errors and try to carry on.

Reply:
Perhaps rebind should directly return `std::io::Error`?  This branch returns `NotConnected` while in `try_send` it returns `BrokenPipe`.  Should these not be consistent?